File: SMS/Forms/Training Assessment.cs
using System;
using System.Collections.Generic;
using System.Data.OleDb;
using System.IO;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace SMS
{
    public partial class TrainingAssessment : Form
    {
        public TrainingAssessment()
        {
            InitializeComponent();
        }

        private void AnalyzeButton_Click(object sender, EventArgs e)
        {
            AssessmentDataGridView.Rows.Clear();
            AssessmentDataGridView.Refresh();

            var selectedSkill = GetSkillName(SkillComboBox.SelectedItem.ToString());

            foreach (var employee in GetEmployeesForTraining(selectedSkill))
                AssessmentDataGridView.Rows.Add(employee);
        }

        private void ReportButton_Click(object sender, EventArgs e)
        {
            try
            {
                var report = new StringBuilder();
                report.AppendLine("Training Assessment Report");
                report.AppendLine(new string('=', "Training Assessment Report".Length));
                report.AppendLine(Environment.NewLine + "Dated: " + DateTime.Today.ToString("MMMM dd, yyyy"));

                foreach (var item in SkillComboBox.Items)
                {
                    report.AppendLine(Environment.NewLine + item);
                    report.AppendLine(new string('-', item.ToString().Length) + Environment.NewLine);

                    var employees = GetEmployeesForTraining(item.ToString());

                    if (employees.Any())
                    {
                        foreach (var employee in employees)
                        {
                            report.AppendLine($"{employee[0]} - {employee[1]}  " +
                                              $"Sex: {employee[2]}  " +
                                              $"Age: {employee[3]}  " +
                                              $"Salary: {employee[4]}");
                        }          
                    }
                    else
                    {
                        report.AppendLine($"No employees require {item} training." + Environment.NewLine);
                    }
                }

                using (var saveFileDialog = new SaveFileDialog())
                {
                    saveFileDialog.Filter = "Text Files (*.txt) | *.txt";
                    saveFileDialog.InitialDirectory = Environment.GetFolderPath(Environment.SpecialFolder.Desktop);

                    if (saveFileDialog.ShowDialog() == DialogResult.OK)
                        File.WriteAllText(saveFileDialog.FileName, report.ToString());

                    MessageBox.Show($"Report successfully exported to {saveFileDialog.FileName}",
                        "Export Successful", MessageBoxButtons.OK, MessageBoxIcon.Information);
                }
            }
            catch (Exception exception)
            {
                MessageBox.Show(String.Format("{0}{1}{1}Details: {2}", "An error occurred.", Environment.NewLine,
                    exception.Message), "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
            }
        }

        private List<object[]> GetEmployeesForTraining(string selectedSkill)
        {
            try
            {
                List<object[]> employeeList = new List<object[]>();
                using (var connection = new OleDbConnection("Provider=Microsoft.ACE.OLEDB.12.0;Data Source=\"Database.accdb\""))
                {
                    connection.Open();
                    using (var row = new OleDbCommand("SELECT * FROM Employees", connection))
                    using (var red = row.ExecuteReader())
                    {
                        while (red != null && red.Read())
                        {
                            if (!Convert.ToBoolean(red[GetSkillName(selectedSkill)]))
                            {
                                employeeList.Add(new object[]
                                {
                                    red["EmployeeID"].ToString(),
                                    red["FullName"].ToString(),
                                    red["Sex"].ToString(),
                                    red["Age"].ToString(),
                                    $"{int.Parse(red["Salary"].ToString()):C2}"
                                });
                            }
                        }
                        return employeeList;
                    }
                }
            }
            catch (Exception exception)
            {
                MessageBox.Show(String.Format("{0}{1}{1}Details: {2}", "An error occurred.", Environment.NewLine,
                    exception.Message), "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
                return null;
            }
        }

        private string GetSkillName(string selectedSkill)
        {
            switch (selectedSkill)
            {
                case "C#":
                    return "CSharp";
                case "C++":
                    return "CPP";
                default:
                    return selectedSkill;
            }
        }

        private void TrainingAssessment_Load(object sender, EventArgs e)
        {
            SkillComboBox.SelectedIndex = 0;
        }
    }
}

File: SMS/Forms/Training Assessment.vb
Imports System.Data.OleDb
Imports System.IO
Imports System.Text

Public Partial Class TrainingAssessment
    Private Sub AnalyzeButton_Click(sender As Object, e As EventArgs) Handles AnalyzeButton.Click
        AssessmentDataGridView.Rows.Clear()
        AssessmentDataGridView.Refresh()

        Dim selectedSkill = GetSkillName(SkillComboBox.SelectedItem.ToString())

        For Each employee As Object() In GetEmployeesForTraining(selectedSkill)
            AssessmentDataGridView.Rows.Add(employee)
        Next
    End Sub

    Private Sub ReportButton_Click(sender As Object, e As EventArgs) Handles ReportButton.Click
        Try
            Dim report = New StringBuilder()
            report.AppendLine("Training Assessment Report")
            report.AppendLine(New String("="C, "Training Assessment Report".Length))
            report.AppendLine(Environment.NewLine + "Dated: " + DateTime.Today.ToString("MMMM dd, yyyy"))

            For Each item As Object In SkillComboBox.Items
                report.AppendLine(Environment.NewLine + item)
                report.AppendLine(New String("-"C, item.ToString().Length) + Environment.NewLine)

                Dim employees = GetEmployeesForTraining(item.ToString())

                If employees.Any() Then
                    For Each employee In employees
                        report.AppendLine(String.Format("{0} - {1}  Sex: {2}  Age: {3}  Salary: {4}", employee(0),employee(1),employee(2), employee(3), employee(4)))
                    Next
                Else
                    report.AppendLine(String.Format("No employees require {0}  training.{1}", item.ToString(),  Environment.NewLine))
                End If
            Next

            Using saveFileDialog = New SaveFileDialog()
                saveFileDialog.Filter = "Text Files (*.txt) | *.txt"
                saveFileDialog.InitialDirectory = Environment.GetFolderPath(Environment.SpecialFolder.Desktop)

                If saveFileDialog.ShowDialog() = DialogResult.OK Then
                    File.WriteAllText(saveFileDialog.FileName, report.ToString())
                        MessageBox.Show("Report successfully exported to " + saveFileDialog.FileName, "Export Successful", MessageBoxButtons.OK, MessageBoxIcon.Information)        
                End If

            End Using
        Catch exception As Exception
            MessageBox.Show(String.Format("{0}{1}{1}Details: {2}", "An error occurred.", Environment.NewLine, exception.Message), "Error", MessageBoxButtons.OK, MessageBoxIcon.[Error])
        End Try
    End Sub

    Private Function GetEmployeesForTraining(selectedSkill As String) As List(Of Object())
        Try
            Dim employeeList As New List(Of Object())()
            Using connection = New OleDbConnection("Provider=Microsoft.ACE.OLEDB.12.0;Data Source=""Database.accdb""")
                connection.Open()
                Using row = New OleDbCommand("SELECT * FROM Employees", connection)
                    Using red = row.ExecuteReader()
                        While red IsNot Nothing AndAlso red.Read()
                            If Not Convert.ToBoolean(red(GetSkillName(selectedSkill))) Then
                                employeeList.Add(New Object() {red("EmployeeID").ToString(), red("FullName").ToString(), 
                                                 red("Sex").ToString(), red("Age").ToString(), 
                                                 String.Format("{0:C2}", Integer.Parse(red("Salary").ToString()))})
                            End If
                        End While
                        Return employeeList
                    End Using
                End Using
            End Using
        Catch exception As Exception
            MessageBox.Show(String.Format("{0}{1}{1}Details: {2}", "An error occurred.", Environment.NewLine, exception.Message), "Error", MessageBoxButtons.OK, MessageBoxIcon.[Error])
            Return Nothing
        End Try
    End Function

    Private Function GetSkillName(selectedSkill As String) As String
        Select Case selectedSkill
            Case "C#"
                Return "CSharp"
            Case "C++"
                Return "CPP"
            Case Else
                Return selectedSkill
        End Select
    End Function

    Private Sub TrainingAssessment_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        SkillComboBox.SelectedIndex = 0
    End Sub
End Class